MobileApplication/MobileApplication/ViewModels/Goal/GoalDetailViewModel.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using MobileApplication.Models;
+using Xamarin.Forms;
+using DbEntities.Entities;
+using MobileApplication.Services.Rest;
+
+namespace MobileApplication.ViewModels
+{
+    class GoalDetailViewModel : BaseViewModel
+    {
+        private readonly IRestService<Goal> RestService;
+        public GoalDetailViewModel(Goal goal)
+        {
+            Id = goal.Id.ToString();
+            Amount = goal.Amount.ToString();
+            Description = goal.Description;
+            Title = goal.Title;
+            DeadlineDate = (DateTime)goal.Deadlinedate;
+            CreationDate = goal.Creationdate;
+            CreationDateString = goal.Creationdate.ToString("dd/MM/yyyy");
+
+            RestService = new RestService<Goal>("api/Goals/");
+
+            SaveCommand = new Command(OnSave, ValidateSave);
+            DeleteCommand = new Command(OnDelete);
+            this.PropertyChanged +=
+                (_, __) => SaveCommand.ChangeCanExecute();
+        }
+
+        public Command SaveCommand { get; }
+        public Command DeleteCommand { get; }
+
+        private string amount;
+        private string description;
+        private string title;
+        private DateTime creationDate;
+        private string creationDateString;
+        private DateTime deadlineDate;
+        public string Id { get; set; }
+
+        public string Amount
+        {
+            get => amount;
+            set => SetProperty(ref amount, value);
+        }
+
+        public string Title
+        {
+            get => title;
+            set => SetProperty(ref title, value);
+        }
+
+        public string Description
+        {
+            get => description;
+            set => SetProperty(ref description, value);
+        }
+
+        public string CreationDateString
+        {
+            get => creationDateString;
+            set => SetProperty(ref creationDateString, value);
+        }
+
+        public DateTime CreationDate
+        {
+            get => creationDate;
+            set => SetProperty(ref creationDate, value);
+        }
+
+        public DateTime DeadlineDate
+        {
+            get => deadlineDate;
+            set => SetProperty(ref deadlineDate, value);
+        }
+
+        private bool ValidateSave()
+        {
+            return !String.IsNullOrEmpty(description)
+                && !String.IsNullOrEmpty(title)
+                && !String.IsNullOrEmpty(amount)
+                && decimal.TryParse(amount, out _)
```

Comment:
What does this underscore do?

Reply:
It's called a discard, it's a placeholder variable that is intentionally unused.